employees/tests/test_unit_report_viewset.py

```diff
@@ -0,0 +1,143 @@
+            description='Some other description',
+            author=other_user,
+            project=self.project,
+            work_hours=Decimal('8.00'),
+        )
+        other_report.full_clean()
+        other_report.save()
+
+        request = APIRequestFactory().get(path=reverse('report-list'))
+        request.user = self.user
+        response = ReportViewSet.as_view({'get': 'list'})(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, other_report.description)
+
+    def test_that_report_list_view_should_add_new_report_on_post(self):
+        request = APIRequestFactory().post(
+            path=reverse('report-list'),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': 'Some description',
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+            }
+        )
+        request.user = self.user
+        response = ReportViewSet.as_view({'post': 'list'})(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 200)
+
+    """
+    -------------
+    REPORT DETAIL
+    -------------
+    """
+    def test_report_detail_view_should_display_report_details(self):
+        request = APIRequestFactory().get(path=reverse('report-detail', args=(self.report.pk,)))
+        request.user = self.user
+        response = ReportViewSet.as_view({'get': 'retrieve'})(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.report.description)
+
+    def test_report_list_view_should_not_be_accessible_for_unauthenticated_users(self):
+        request = APIRequestFactory().get(path=reverse('report-detail', args=(self.report.pk,)))
+        request.user = AnonymousUser()
+        response = ReportViewSet.as_view({'get': 'retrieve'})(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 403)
+
+    def test_report_detail_view_should_not_render_non_existing_report(self):
+        request = APIRequestFactory().get(path=reverse('report-detail', args=(999,)))
+        request.user = self.user
+        response = ReportViewSet.as_view({'get': 'retrieve'})(request, pk=999)
+        self.assertEqual(response.status_code, 404)
+
+    def test_report_detail_view_should_update_report_on_put(self):
+        request = APIRequestFactory().put(
+            path=reverse('report-detail', args=(self.report.pk,)),
+            data={
+                'date': datetime.datetime.now().date(),
+                'description': 'Some other description',
+                'project': self.project,
+                'work_hours': Decimal('8.00'),
+            }
+        )
+        request.user = self.user
+        response = ReportViewSet.as_view({'put': 'retrieve'})(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 200)
+
+    def test_report_detail_view_should_delete_report_on_delete(self):
```

Comment:
Same as above.

Reply:
Also check the database.

Reply:
And here, after I applied changes in method call, I can also change HTTP request from 'delete' to (if I remember correctly) 'post' and 'get' and I still get the desired result.

Makes little sense to me, but I guess that's just how REST operates?

Reply:
Yes, see response above.

Reply:
Done